contracts/vesting/VestingBase.sol

```diff
@@ -0,0 +1,145 @@
+    ///@return Returns the closing balance of vesting coin held by this contract.
+    function getAvailableFunds() public view returns (uint256) {
+        return vestingCoin.balanceOf(address(this));
+    }
+
+    ///@notice Gets the markle tree of each vesting Rounds.
+    ///@param _round The round of which markleTree to be viewed.
+    ///@return Returns Total vested balance.
+    function getMerkleRoot(uint256 _round)
+        public
+        view
+        onlyOwner
+        returns (bytes32)
+    {
+        return RootToRounds[_round];
+    }
+
+    ///@notice Enables this vesting contract to receive the ERC20 (vesting coin).
+    ///Before calling this function please approve your desired amount of the coin
+    ///for this smart contract address.
+    ///Please note that this action is restricted to administrators only.
+    ///@return Returns true if the funding was successful.
+    function fund() external onlyOwner returns (bool) {
+        ///Check the funds available.
+        uint256 allowance = vestingCoin.allowance(msg.sender, address(this));
+        require(allowance > 0, "Nothing to fund.");
+
+        ///Get the current allocation.
+        uint256 current = getAvailableFunds();
+
+        require(vestingCoin.transferFrom(msg.sender, address(this), allowance));
+
+        emit Funded(msg.sender, allowance, current, getAvailableFunds());
+        return true;
+    }
+
+    ///@notice Allows you to withdraw the surplus balance of the vesting coin from this contract.
+    ///Please note that this action is restricted to administrators only
+    ///and you may only withdraw amounts above the sum total allocation balances.
+    ///@param _amount The amount desired to withdraw.
+    ///@return Returns true if the withdrawal was successful.
+    function removeFunds(uint256 _amount) external onlyOwner returns (bool) {
+        uint256 balance = vestingCoin.balanceOf(address(this));
+
+        uint256 available = balance - totalWithdrawn;
+
+        require(available >= _amount, "amount is grater than balanace");
+
+        require(vestingCoin.transfer(msg.sender, _amount));
+
+        emit FundRemoved(msg.sender, _amount, available.sub(_amount));
+        return true;
+    }
+
+    ///@notice This action enables admin to set newMarkelRoot.
+
+    function setMerkleRoot(bytes32 _newMerkleRoot, uint _round) external onlyOwner {
+        require(
+            _newMerkleRoot != 0x00,
+            "Vesting: Invalid new merkle root value!"
+        );
+
+    
+        RootToRounds[_round] = _newMerkleRoot;
+    }
+
+    ///@notice This action enables the beneficiaries to withdraw a desired amount from this contract.
+    ///@param _amount The amount in vesting coin desired to withdraw.
+    function withdraw(
+        uint256 _amount,
+        bytes32[] calldata _proof,
+        uint256 _round
+    ) external whenNotPaused returns (bool) {
+        require(
+            !vestingClaimed[msg.sender],
```

Comment:
Let's make `vestingClaimed` a mapping that have structure like this
```js
vestingClaimed[_round][msg.sender]
```
This way it will check whether per round those who have claim have claimed or not. Otherwise, this will be limited only by one withdrawal and the rest will revert

Reply:
Done